R/utils.R

```diff
@@ -158,15 +158,22 @@ reset_lang <- function(old_lang) {
 #' @param fun A function that takes a source file and returns `lint` objects.
 #' @param name Default name of the Linter.
 #' Lints produced by the linter will be labelled with `name` by default.
+#' @param linter_level Which level of expression is the linter working with?
+#'   `"expression"` means an individual expression in `xml_parsed_content`, while `"file"` means all expressions
+#'   in the current file are available in `full_xml_parsed_content`.
+#'   `NA` means the linter will be run with both, expression-level and file-level source expressions.
+#'
 #' @return The same function with its class set to 'linter'.
 #' @export
-Linter <- function(fun, name = linter_auto_name()) { # nolint: object_name.
+Linter <- function(fun, name = linter_auto_name(), linter_level = c(NA_character_, "file", "expression")) { # nolint: object_name, line_length.
```

Comment:
Nit: Instead of turning off the line length linter, why not just have one argument per line here?

Reply:
I felt it looked less cluttered here because all that exceeds the line are nolint statements.
I worry that a `# nolint next` would mess up roxygen here. Maybe time for a `# nolint prev`? 😁

Reply:
> I worry that a # nolint next would mess up roxygen here. 

It wouldn't, plain comments are simply skipped:

> Blocks ignore empty lines, including lines made up of non-roxygen comments.

https://roxygen2.r-lib.org/articles/rd.html#basics

Reply:
Good to know, thanks!